### PR TITLE
Managed Dependencies is not supported on Legion

### LINKED ISSUE #944: Managed Dependencies is not supported on Legion
We should error out when a user tries to use Managed Dependencies on Legion.

### PR DESCRIPTION
**Pull request by Francisco-Gamino**
### Issue describing the changes in this PR

Resolves https://github.com/Azure/azure-functions-powershell-worker/issues/944

### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [ ] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: src/DependencyManagement/ManagedDependenciesPathDetector.cs [hunk 1]
```diff
@@ -30,6 +31,11 @@ internal static class ManagedDependenciesPathDetector
         /// </summary>
         public static string GetManagedDependenciesPath(string functionAppRootPath)
         {
+            if (IsLinuxConsumptionOnLegion())
```
Comment (AnatoliB) [inline src/DependencyManagement/ManagedDependenciesPathDetector.cs]: Consider moving this check to a more intuitively obvious place, such as: https://github.com/Azure/azure-functions-powershell-worker/blob/25260ecd0b3f5fdb1e3a9297b7b80ec26245f80c/src/DependencyManagement/DependencyManager.cs#L107

This error should happen _only_ when the `requirements.psd1` contains any module references. The error should _not_ happen when `requirements.psd1` is absent, empty, or contains no module references. For example, a brand new app generated from a template should _not_ report this error. Chances are `GetManagedDependenciesPath` invocation does  not depend on the same factors. I suggest performing this check in `DependencyManager.Initialize`, just after we made sure that `_dependenciesFromManifest` is not empty.

File: src/resources/PowerShellWorkerStrings.resx [hunk 2]
```diff
@@ -376,4 +376,7 @@
   <data name="UtilizingExternalDurableSDK" xml:space="preserve">
     <value>Utilizing external Durable Functions SDK: '{0}'.</value>
   </data>
+  <data name="ManagedDependenciesIsNotSupportedOnLegion" xml:space="preserve">
+    <value>Managed Dependencies is not supported in Linux Consumption on Legion. Please remove all module references from requirements.psd1 and include the function app dependencies with the function app content. For more information, please see https://aka.ms/functions-powershell-include-modules-with-app-content.</value>
```
Comment (AnatoliB) [inline src/resources/PowerShellWorkerStrings.resx]: Consider reducing the name to just `functions-powershell-include-modules` - much shorter, but still unambiguously expressing the intent.

File: src/DependencyManagement/ManagedDependenciesPathDetector.cs (hunk 1 above)
Comment (Francisco-Gamino) [reply src/DependencyManagement/ManagedDependenciesPathDetector.cs]: Thank you @AnatoliB for the suggestion. I have updated the code accordingly.

File: src/resources/PowerShellWorkerStrings.resx (hunk 2 above)
Comment (Francisco-Gamino) [reply src/resources/PowerShellWorkerStrings.resx]: Great suggestion. Fixed.

File: src/DependencyManagement/DependencyManager.cs [hunk 3]
```diff
@@ -99,6 +99,11 @@ internal string Initialize(ILogger logger)
                 // Parse and process the function app dependencies defined in the manifest.
                 _dependenciesFromManifest = _storage.GetDependencies().ToArray();
 
+                if (_dependenciesFromManifest.Length >= 1 && WorkerEnvironment.IsLinuxConsumptionOnLegion())
```
Comment (AnatoliB) [inline src/DependencyManagement/DependencyManager.cs]: If you put this block under `if (!_dependenciesFromManifest.Any()) { ... }`, you will also be able to remove the first part of the condition.

File: test/Unit/DependencyManagement/DependencyManagerTests.cs [hunk 4]
```diff
@@ -50,6 +50,38 @@ public void Initialize_ReturnsNewSnapshotPath_WhenNoAcceptableDependencyVersions
             }
         }
 
+        [Fact]
+        public void Initialize_Throws_WhenDependenciesAreDefinedInRequirementsPsd1_OnLegion()
```
Comment (AnatoliB) [inline test/Unit/DependencyManagement/DependencyManagerTests.cs:54]: Consider adding a test that verifies that we _don't_ throw an exception on Legion when the dependency manifest is empty

File: src/DependencyManagement/DependencyManager.cs (hunk 3 above)
Comment (Francisco-Gamino) [reply src/DependencyManagement/DependencyManager.cs]: Good point. Fixed.

File: test/Unit/DependencyManagement/DependencyManagerTests.cs (hunk 4 above)
Comment (Francisco-Gamino) [reply test/Unit/DependencyManagement/DependencyManagerTests.cs:54]: Test case added.

**Review by AnatoliB — APPROVED** [review]
LGTM

**Comment by AnatoliB** [comment]
@Francisco-Gamino I think this needs to be backported to 7.2
